bugsnag-android-core/src/main/java/com/bugsnag/android/Configuration.java

```diff
@@ -0,0 +1,821 @@
+     *
+     * To override this behavior, an alternative instance can be provided that implements the
+     * Logger interface.
+     */
+    public void setLogger(@Nullable Logger logger) {
+        impl.setLogger(logger);
+    }
+
+    /**
+     * The Delivery implementation used to make network calls to the Bugsnag
+     * <a href="https://docs.bugsnag.com/api/error-reporting/">Error Reporting</a> and
+     * <a href="https://docs.bugsnag.com/api/sessions/">Sessions API</a>.
+     *
+     * This may be useful if you have requirements such as certificate pinning and rotation,
+     * which are not supported by the default implementation.
+     *
+     * To provide custom delivery functionality, create a class which implements the Delivery
+     * interface. Please note that request bodies must match the structure specified in the
+     * <a href="https://docs.bugsnag.com/api/error-reporting/">Error Reporting</a> and
+     * <a href="https://docs.bugsnag.com/api/sessions/">Sessions API</a> documentation.
+     *
+     * You can use the return type from the deliver functions to control the strategy for
+     * retrying the transmission at a later date.
+     *
+     * If DeliveryStatus.UNDELIVERED is returned, the notifier will automatically cache
+     * the payload and trigger delivery later on. Otherwise, if either DeliveryStatus.DELIVERED
+     * or DeliveryStatus.FAILURE is returned the notifier will removed any cached payload
+     * and no further delivery will be attempted.
+     */
+    @NonNull
+    public Delivery getDelivery() {
+        return impl.getDelivery();
+    }
+
+    /**
+     * The Delivery implementation used to make network calls to the Bugsnag
+     * <a href="https://docs.bugsnag.com/api/error-reporting/">Error Reporting</a> and
+     * <a href="https://docs.bugsnag.com/api/sessions/">Sessions API</a>.
+     *
+     * This may be useful if you have requirements such as certificate pinning and rotation,
+     * which are not supported by the default implementation.
+     *
+     * To provide custom delivery functionality, create a class which implements the Delivery
+     * interface. Please note that request bodies must match the structure specified in the
+     * <a href="https://docs.bugsnag.com/api/error-reporting/">Error Reporting</a> and
+     * <a href="https://docs.bugsnag.com/api/sessions/">Sessions API</a> documentation.
+     *
+     * You can use the return type from the deliver functions to control the strategy for
+     * retrying the transmission at a later date.
+     *
+     * If DeliveryStatus.UNDELIVERED is returned, the notifier will automatically cache
+     * the payload and trigger delivery later on. Otherwise, if either DeliveryStatus.DELIVERED
+     * or DeliveryStatus.FAILURE is returned the notifier will removed any cached payload
+     * and no further delivery will be attempted.
+     */
+    public void setDelivery(@NonNull Delivery delivery) {
```

Comment:
It wouldn't make sense for users to set `Delivery` to null as no payloads would be delivered to our API. It is nullable in Kotlin as an implementation detail - we can't instantiate `DefaultDelivery` in `Configuration` because it requires a `Context`. This context object is supplied when a user initialises the notifier, so we then instantiate `Delivery` if nothing else has been set.